### PR TITLE
refactor(*): Switch from arrow function properties to @autoBindMethods decorator

### PR DESCRIPTION
**Pull request by christopherthielen**

### CODE REVIEW

File: app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx [hunk 1]
```diff
@@ -144,18 +146,18 @@ export class ExecutionGroup extends React.Component<IProps, IState> {
     }
   }
 
-  private handleHeadingClicked = () => {
+  private handleHeadingClicked(): void {
```
Comment (christopherthielen) [inline app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx]: FWIW, I find adding `: void` return types tedious and mostly unnecessary

**Comment by christopherthielen** [comment]
I spot checked the app and things didn't seem broken

**Comment by jrsquared** [comment]
Oh nice, I didn't see/try `class-autobind-decorator`. Glad you found one that worked!!

File: app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx (hunk 1 above)
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx]: I agree! I will stand strong with you and not add void return types to methods.

**Comment by jrsquared** [comment]
LGTM. I didn't smoke it but looks good...

File: app/scripts/modules/core/delivery/create/CreatePipelineModal.tsx [hunk 2]
```diff
@@ -112,40 +115,40 @@ export class CreatePipelineModal extends React.Component<ICreatePipelineModalPro
...
+  }
 
-  private handleErrorDismiss = (): void => {
+  private handleErrorDismiss(): void {
     this.setState({saveError: false});
-  };
+  }
 
-  private templateOptionRenderer = (option: Select.Option) => {
+  private templateOptionRenderer(option: Select.Option) {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/create/CreatePipelineModal.tsx]: I'm on board with omitting return types of void, but this returns `React.Element[]` or something, right?

(I know you were only updating the autobind bits, just noticed this now and while you're in there...)

File: app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx (hunk 1 above)
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx]: The only upside of adding `: void` return types is it makes it more clear when methods that return something else are missing a return type, so I'm having mixed feelings about it...

File: app/scripts/modules/core/delivery/triggers/NextRunTag.tsx [hunk 3]
```diff
@@ -9,13 +10,14 @@ import { timestamp } from 'core/utils/timeFormatters';
...
 interface IProps {
   pipeline: IPipeline;
-};
+}
 
 interface IState {
   nextScheduled: number;
   hasNextScheduled: boolean;
-};
+}
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/delivery/triggers/NextRunTag.tsx]: I feel like our linter could be helping us more than it is...

Comment (jrsquared) [inline app/scripts/modules/core/delivery/triggers/NextRunTag.tsx]: Haha, the irony of the tslint semicolon rule... enforce semicolons on interface, or ignore linting semicolons on interfaces altogether.

File: app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx (hunk 1 above)
Comment (jrsquared) [inline app/scripts/modules/core/delivery/executionGroup/ExecutionGroup.tsx]: Ah.... that is a good point... So much for standing strong.

File: app/scripts/modules/core/delivery/triggers/NextRunTag.tsx (hunk 3 above)
Comment (christopherthielen) [inline app/scripts/modules/core/delivery/triggers/NextRunTag.tsx]: looks like they have an interface toggle: https://palantir.github.io/tslint/rules/semicolon/
> "ignore-interfaces" skips checking semicolons at the end of interface members.

Comment (jrsquared) [inline app/scripts/modules/core/delivery/triggers/NextRunTag.tsx]: Yeah... 'skips' being the operative word.

Comment (christopherthielen) [inline app/scripts/modules/core/delivery/triggers/NextRunTag.tsx]: Oh drat

**Pull request merged**